changefeedccl: Support more stable functions.

miretskiy opened the pull request:
Add support for additional stable functions to CDC expressions.

Fixes #83466

Release Notes: None

Commit:
changefeedccl: Support more stable functions.

Add support for additional stable functions to CDC expressions.

Fixes #83466

Release Notes: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/83548)
<!-- Reviewable:end -->

HonoreDB reviewed (approved):
Amazing!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/83548#-:-N5gQGTDA9SE-vmL1BEj:b5uixe1)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

miretskiy commented:
fyi @HonoreDB: since I changed function resolution code somewhat, i also added a test
that verifies we get an error when invoking cdc functions with wrong arguments.

miretskiy commented:
bors r+
